### PR TITLE
Update to version of Lightgbm with no runtime dependency on GCC.

### LINKED ISSUE #1067: Update LightGBM nuget so that there is no dependency on GCC
Currently lightgbm nuget depends on a binary for openmp that comes from gcc compiler but having to install gcc compiler on build machines causes slowdown. Update the nuget to the newest version that removes this dependency because lightgbm native code is compiled using clang.

### PR DESCRIPTION
**Pull request by montebhoover**
Update to version of Lightgbm with no runtime dependency on GCC.

Requires adding installation of libomp to Mac build phase. (It now has a runtime dependency on the standard OpenMP (libomp) instead of the GCC implementation (libgomp)).

Fixes #1067 and #494.

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
Do you plan to close @codemzs  PR?

File: test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj [hunk 1]
```diff
@@ -22,6 +22,7 @@
     <ProjectReference Include="..\..\src\Microsoft.ML.Legacy\Microsoft.ML.Legacy.csproj" />
     <ProjectReference Include="..\Microsoft.ML.TestFramework\Microsoft.ML.TestFramework.csproj" />
     <ProjectReference Include="..\Microsoft.ML.Tests\Microsoft.ML.Tests.csproj" />
+    <PackageReference Include="LightGBM" Version="$(LightGBMPackageVersion)" />
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj]: ><PackageReference Include="LightGBM" Version="$(LightGBMPackageVersion)" /> [](start = 4, length = 75)

Don't we pull this dependency through ProjectReference to LightGBM ?

**Comment by montebhoover** [comment]
@Ivanidzo4ka Yes, I will close #1066. My apologies for the mistake there.

File: test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj (hunk 1 above)
Comment (montebhoover) [reply test/Microsoft.ML.Core.Tests/Microsoft.ML.Core.Tests.csproj]: You're right, I see that we do.  I'll remove the unnecessary PackageReferences.

File: build/ci/phase-template.yml [hunk 2]
```diff
@@ -19,6 +19,9 @@ phases:
     steps:
     - script: $(_buildScript) -$(_configuration)
       displayName: Build
+    - ${{ if eq(parameters.name, 'MacOS') }}:
+      - script: brew install libomp
```
[the remark this answers is not in the document]
Comment (montebhoover) [reply build/ci/phase-template.yml]: Sounds good.  I can make a quick PR for that change.

File: test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj [hunk 3]
```diff
@@ -14,6 +14,7 @@
     <ProjectReference Include="..\..\src\Microsoft.ML.HalLearners\Microsoft.ML.HalLearners.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.ML.ImageAnalytics\Microsoft.ML.ImageAnalytics.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.ML.KMeansClustering\Microsoft.ML.KMeansClustering.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.ML.LightGBM\Microsoft.ML.LightGBM.csproj" />
```
[the remark this answers is not in the document]
Comment (montebhoover) [reply test/Microsoft.ML.Tests/Microsoft.ML.Tests.csproj]: You're right, we don't need the LightGBM reference in this project.  I'll remove that in the PR to move to Hosted macOS.

**Pull request merged**